src/mlpack/methods/cf/cf_model_impl.hpp

```diff
@@ -105,15 +118,49 @@ void CFModel::Train(const MatType& data,
                     const size_t rank,
                     const size_t maxIterations,
                     const double minResidue,
-                    const bool mit)
+                    const bool mit,
+                    const std::string& normalization)
 {
   // Delete the current CFType object, if there is one.
   boost::apply_visitor(DeleteVisitor(), cf);
 
   // Instantiate a new CFType object.
   DecompositionPolicy decomposition;
-  cf = new CFType<DecompositionPolicy>(data, decomposition,
-      numUsersForSimilarity, rank, maxIterations, minResidue, mit);
+  if (normalization == "overallMean")
+  {
+    cf = new CFType<DecompositionPolicy, OverallMeanNormalization>(data,
+        decomposition, numUsersForSimilarity, rank, maxIterations, minResidue,
+        mit);
+  }
+  else if (normalization == "itemMean")
+  {
+    cf = new CFType<DecompositionPolicy, ItemMeanNormalization>(data,
+        decomposition, numUsersForSimilarity, rank, maxIterations, minResidue,
+        mit);
+  }
+  else if (normalization == "userMean")
+  {
+    cf = new CFType<DecompositionPolicy, UserMeanNormalization>(data,
+        decomposition, numUsersForSimilarity, rank, maxIterations, minResidue,
+        mit);
+  }
+  else if (normalization == "zScore")
+  {
+    cf = new CFType<DecompositionPolicy, ZScoreNormalization>(data,
+        decomposition, numUsersForSimilarity, rank, maxIterations, minResidue,
+        mit);
+  }
+  else if (normalization == "none")
+  {
+    cf = new CFType<DecompositionPolicy, NoNormalization>(data,
+        decomposition, numUsersForSimilarity, rank, maxIterations, minResidue,
+        mit);
+  }
+  else
+  {
+    throw std::runtime_error("Unsupported Normalization Algorithm."
+                             "Please refer to help (-h) section for details");
```

Comment:
Actually we should remove the `(-h)` here, since this binding's output will be displayed for all languages, even Python and Julia where `-h` don't make sense.  Honestly I'd avoid referencing the documentation here and write the list of supported normalization choices.  If we just point to the documentation, that's another step the user has to take before they successfully solve their problem. :)

Reply:
Yes, it seems a lot more reasonable. Will add the supported CLI normalizations. :+1: